pkg/providers/az-client.go

```diff
@@ -43,3 +46,36 @@ func (g *GraphServiceClient) GetApplicationObjectId(ctx context.Context, appId s
 	}
 	return *appObjectId, nil
 }
+
+type RoleAssignmentClient struct {
+	Client *armauthorization.RoleAssignmentsClient
+}
+
+type RoleAssignClient interface {
+	CreateRoleAssignment(ctx context.Context, objectId, roleId, scope, raUid string) error
+}
+
+var _ RoleAssignClient = &RoleAssignmentClient{}
+
+func (r *RoleAssignmentClient) CreateRoleAssignment(ctx context.Context, objectId, roleId, scope, raUid string) error {
+	log.Debug("Assigning contributor role to service principal...", "objectId", objectId, "role assignment UID", raUid, "scope", scope)
+
+	fullAssignmentId := fmt.Sprintf("/%s/providers/Microsoft.Authorization/roleAssignments/%s", scope, raUid)
+	fullDefinitionId := fmt.Sprintf("/providers/Microsoft.Authorization/roleDefinitions/%s", roleId)
+
+	params := armauthorization.RoleAssignmentCreateParameters{
+		Properties: &armauthorization.RoleAssignmentProperties{
+			PrincipalID:      &objectId,
```

Comment:
PrincipalType is also missed from the Properties - "--assignee-principal-type", "ServicePrincipal" , previous az command had this type which can set using 
```PrincipalType:    to.Ptr(armauthorization.PrincipalTypeUser)```

Reply:
I don't see PrincipalType in the RoleAssignementProperties, but you are right its present in the CLI calls, I should see if that's the required property and if so have to look for an alternative method.
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization@v1.0.0#section-readme